backend/src/db/services/lastwills.service.ts

```diff
@@ -0,0 +1,78 @@
+import { InjectModel } from '@m8a/nestjs-typegoose'
+import {
+  Injectable,
+  Logger,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common'
+import { ReturnModelType } from '@typegoose/typegoose'
+import { ObjectId } from 'mongoose'
+import { CreateLastWillDto } from '../../last-wills/dto/create-lastwill.dto'
+import { UpdateLastWillDto } from '../../last-wills/dto/update-lastwill.dto'
+import { paymentPlans } from '../../payments/interfaces/payments'
+import { LastWill } from '../entities/lastwill.entity'
+import { UserService } from './user.service'
+
+@Injectable()
+export class LastWillsService {
+  private readonly logger = new Logger(LastWillsService.name)
+  constructor(
+    @InjectModel(LastWill)
+    private readonly lastWillModel: ReturnModelType<typeof LastWill>,
+    private readonly userService: UserService,
+  ) {}
+
+  async createOne(createLastWillDto: CreateLastWillDto, userId: ObjectId) {
+    const lastWillCount = await this.lastWillModel.countDocuments({
```

Comment:
Move the validation of this to a separate service within the lastwillmodule

This service should expose two functions based on the needed functionality for this:

```js
getLastWillCountByUserId(id): number

createLastWill(userId, lastWilldata): LastWill
```

The fact that this service depends on another service is a clear indicator that there is too much going on in here